lib/galaxy_test/selenium/test_admin_app.py

```diff
@@ -8,6 +9,46 @@ class AdminAppTestCase(SeleniumTestCase):
 
     requires_admin = True
 
+    @selenium_test
+    @flakey  # This test relies on an external server that may or may not be available to the testing environment.
+    def test_admin_toolshed(self):
+        admin_component = self.components.admin
+        self.admin_login()
+        self.admin_open()
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.screenshot('admin_landing')
+        admin_component.index.toolshed.wait_for_and_click()
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.screenshot('admin_toolshed_landing')
+        repo_search_input = self.driver.find_element_by_id('toolshed-repo-search')
+        repo_search_input.clear()
+        repo_search_input.send_keys('all_fasta')
+        # If this hasn't succeeded after 30 seconds, the @flakey context should
+        # allow the test to still pass, since there should definitely be results
+        # after 30 seconds.
+        self.sleep_for(self.wait_types.SHED_SEARCH)
+        self.screenshot('admin_toolshed_search')
+        admin_component.toolshed.search_results.wait_for_visible()
+        # It's very unlikely that this data manager will ever stop existing.
+        repository_row = self.driver.find_element_by_link_text('data_manager_fetch_genome_dbkeys_all_fasta')
+        repository_row.click()
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.screenshot('admin_toolshed_repo_details')
+        install_button = self.driver.find_element_by_xpath("(//button[contains(., 'Install')])[1]")
+        install_button.click()
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.screenshot('admin_toolshed_repo_install_settings')
+        ok_button = self.driver.find_element_by_xpath("//*[@id='repo-install-settings___BV_modal_footer_']/button[contains(., 'OK')]")
```

Comment:
Can you put these in navigation.yml?

If instead of a string for the selector you do a dict - you can specify the type as xpath. https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/selenium/navigation.yml#L34.

```yml
    user:
      type: xpath
      selector: '//li[@id="user"]'
```

Better yet though if you could find where we are rendering the button and give it a cleaner class? 'repo-settings-install-ok', 'repo-settings-uninstall'.